verification/curator-service/ui/src/components/new-case-form-fields/Events.tsx

```diff
@@ -36,90 +85,60 @@ const methodsOfConfirmation = [
-                        {this.dateField('confirmedDate', 'Confirmed case date')}
-                        {this.selectField(
-                            'methodOfConfirmation',
-                            'Method of confirmation',
-                            methodsOfConfirmation,
-                        )}
-                        {this.dateField(
-                            'onsetSymptomsDate',
-                            'Onset of symptoms date',
-                        )}
-                        {this.dateField(
-                            'firstClinicalConsultationDate',
-                            'First clinical consultation date',
-                        )}
-                        {this.dateField(
-                            'selfIsolationDate',
-                            'Self isolation date',
-                        )}
-                        {this.dateField(
-                            'hospitalAdmissionDate',
-                            'Hospital admission date',
-                        )}
-                        {this.dateField(
-                            'icuAdmissionDate',
-                            'ICU admission date',
-                        )}
-                        {this.dateField('outcomeDate', 'Outcome date')}
-                        {this.selectField('outcome', 'Outcome', outcomes)}
-                    </MuiPickersUtilsProvider>
-                </fieldset>
-            </Scroll.Element>
-        );
-    }
+export default function Events(): JSX.Element {
+    const { values } = useFormikContext();
+    return (
+        <Scroll.Element name="events">
+            <fieldset>
+                <legend>Events</legend>
+                <MuiPickersUtilsProvider utils={DateFnsUtils}>
+                    <DateField
+                        name="confirmedDate"
+                        label="Confirmed case date"
+                    ></DateField>
+                    <SelectField
+                        name="methodOfConfirmation"
+                        label="Method of confirmation"
+                        values={methodsOfConfirmation}
+                    ></SelectField>
+                    <DateField
+                        name="onsetSymptomsDate"
+                        label="Onset of symptoms date"
+                    ></DateField>
+                    <DateField
+                        name="firstClinicalConsultationDate"
+                        label="First clinical consultation date"
+                    ></DateField>
+                    <DateField
+                        name="selfIsolationDate"
+                        label="Self isolation date"
+                    ></DateField>
+                    <SelectField
+                        name="admittedToHospital"
+                        label="Hospital admission"
+                        values={yesNoUndefined}
+                    ></SelectField>
+                    {(values as any).admittedToHospital === 'Yes' && (
```

Comment:
maybe leave a todo that we should export our values interface defined in NewCaseForm to be able to use it here and in other places?
useFormikContext takes a type argument to support that: `useFormikContext<SomeValuesInterface>()`

Reply:
Went ahead and did it in this PR by moving the form values interface into its own file.